pkg/gateway/model_build_targets_test.go

```diff
@@ -65,25 +66,71 @@ func Test_Targets(t *testing.T) {
 					DeletionTimestamp: nil,
 				},
 			},
-			inDataStore:        true,
-			refByServiceExport: true,
-			wantErrIsNil:       true,
+			inDataStore:  true,
+			refByService: true,
+			wantErrIsNil: true,
 			expectedTargetList: []latticemodel.Target{
 				{
 					TargetIP: "10.10.1.1",
 					Port:     8675,
 				},
 				{
-					TargetIP: "10.10.1.1",
-					Port:     309,
+					TargetIP: "10.10.2.2",
+					Port:     8675,
 				},
+			},
+		},
+		{
+			name:               "Add endpoints with matching service port to build spec",
+			srvExportName:      "export1",
+			srvExportNamespace: "ns1",
+			port:               80,
+			endPoints: []corev1.Endpoints{
 				{
-					TargetIP: "10.10.2.2",
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns1",
+						Name:      "export1",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{{IP: "10.10.1.1"}, {IP: "10.10.2.2"}},
+							Ports:     []corev1.EndpointPort{{Name: "a", Port: 8675}, {Name: "b", Port: 309}},
+						},
+					},
+				},
+			},
+			svc: corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:         "ns1",
+					Name:              "export1",
+					DeletionTimestamp: nil,
+				},
+				Spec: corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{
+						{
+							Name:       "a",
+							Port:       80,
+							TargetPort: intstr.FromInt(8675),
+						},
+						{
+							Name:       "b",
+							Port:       81,
+							TargetPort: intstr.FromInt(309),
+						},
+					},
+				},
+			},
+			inDataStore:  true,
+			refByService: true,
+			wantErrIsNil: true,
+			expectedTargetList: []latticemodel.Target{
+				{
+					TargetIP: "10.10.1.1",
 					Port:     8675,
 				},
 				{
 					TargetIP: "10.10.2.2",
-					Port:     309,
+					Port:     8675,
```

Comment:
Can you explain please why it's 8675?

As I understand from code that if port is not named then it's the only port. Here we have 2 named ports "a" and "b" and both targets are generated with "a" port, where second port go?

Reply:
in line 87 - we pick the port 80, this is part of target model build task which is supposed to be filled in from route backendRef

Reply:
should {Name: "b", Port: 309}} be in expected target list?

Reply:
no - by picking 80 in model build task, we know that only relevant service port is {Name: "a", Port: 80, TargetPort: 8675}. The endpoints with other service ports will be ignored.

Reply:
Not sure if you intended to put the same port, but it should not be possible by k8s spec

Reply:
```
core1.ServicePort{ {Name: "a", Port: 80, Target: "8675"}, {Name: "b", Port: 80, Target: "309"} }

```
Seems in that scenario, only one (Port,TargetPort ) can take effect
![image](https://github.com/aws/aws-application-networking-k8s/assets/32318664/f4b214e2-9026-4b48-a30c-5c537020a547)

Reply:
core1.ServicePort{ {Name: "a", Port: 80, Target: "8675"}, {Name: "b", Port: 8090, Target: "309"} }

For this one it can have two (Port,TargetPort )
![image](https://github.com/aws/aws-application-networking-k8s/assets/32318664/50df2bc6-856c-4f4e-ae5a-ecdd5f6d2b3d)

Reply:
by k8s spec it is stored as a map with port as a key, probably that's why.